### PR TITLE
#161779783 Create user account

### PR DESCRIPTION
**Pull request by Daymorelah**
#### What does this PR do?
Adds a feature that allows a user to create an account on the app.
#### Description of Task to be completed?
*  Build controller to handle a **POST** request to the `/api/v1/auth/signup` route
*  Write tests for the controller
*  Update travis.yml file to run tests using sequelize
#### How should this be manually tested?
* Fork the repo and `cd` into the project's directory
* Run `npm install` install all required packages
* Run 'npm migrate` to run migrations
* Run `npm start` to start the app
* Run `npm test` to test the app
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#161779783
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
-install: npm install
-script: npm test
+  - "6"
+env: 
+  global:
+    - NODE_ENV=test
+branches:
+  except:
+  - gh-pages
+  - master
```
Comment (kingisaac95) [inline .travis.yml]: Why are you ignoring `master`?

File: package.json [hunk 2]
```diff
@@ -5,15 +5,18 @@
...
     "heroku-postbuild": "npm run create-database-table",
-    "migrate": "./node_modules/.bin/babel-node ./node_modules/.bin/sequelize db:migrate"
+    "nyc:report": "nyc report",
+    "migrate": "export NODE_ENV=test&& ./node_modules/.bin/babel-node ./node_modules/.bin/sequelize db:migrate",
+    "undo:migration": "export NODE_ENV=test&& ./node_modules/.bin/babel-node ./node_modules/.bin/sequelize db:migrate:undo:all"
   },
   "author": "Andela Simulations Programme",
   "license": "MIT",
   "dependencies": {
+    "bcrypt": "^3.0.2",
```
Comment (kingisaac95) [inline package.json]: lock version

File: test/index.js [hunk 3]
```diff
@@ -1 +1,2 @@
-import './sampleTest';
+import './sample.test';
+import './users.test';
```
Comment (kingisaac95) [inline test/index.js]: new line here

Comment (Daymorelah) [reply test/index.js]: I have added a new line there.

File: package.json (hunk 2 above)
Comment (Daymorelah) [reply package.json]: I have locked the version.

File: .travis.yml (hunk 1 above)
Comment (Daymorelah) [reply .travis.yml]: Since we are not pushing to master branch directly, I decided to prevent Travis form watching for any activity on the master branch

**Review by kingisaac95 — APPROVED** [review]

**Pull request merged**